js/components/search.js

```diff
@@ -5,85 +5,166 @@ $( document ).ready(function() {
+   * @param {number} [options.from=0] - Offset to start from for paging
+   * @returns {string} URL for desired search
+   */
+  function apiSearchUrl( query, apiKey, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return (
+      'https://api.data.gov/beckley-federalist/v0/resources/eiti/?' +
+      'q=' + q + '&size=' + size + '&from=' + from + '&api_key=' + apiKey
+    );
+  }
+
+  function localSearchUrl( query, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return '/search-results/?q=' + q + '&size=' + size + '&from=' + from;
   }
-  else {
+
+  /**
+   * Returns HTML string of Glyph for given content type
+   *
+   * @param {string} contentType - given document content type
+   * @returns {string} glyph and label for content type or generic default
+   */
+  function getContentTypeGlyph(contentType) {
+    var contentTypes = {
+      'text/html':       '<span class="glyphicon glyphicon-link"></span> Website',
+      'application/pdf': '<span class="glyphicon glyphicon-file"></span> PDF',
+      'default':         '<span class="glyphicon glyphicon-file"></span> ' + contentType
+    };
+
+    return contentTypes[contentType]
+      ? contentTypes[contentType]
+      : contentTypes['default']
+  }
+
+  var query = getQueryParam('q'),
+      size  = parseInt(getQueryParam('size'), 10) || 10,
+      from  = parseInt(getQueryParam('from'), 10) || 0;
+
+  $('.search-string').text( query );
+  $('.site-search-text').attr('value', query );
+
+  if ( ! query ) {
     $('.search-no-results').show();
     $('.loading').remove();
     $('.search-results-count').append( 0 + ' search results');
-
+    return;
   }
+
+  $('input.q').val( query );
+  $('.search-result-list').html(
+    '<div class="loading">' +
+      '<span class="glyphicon glyphicon-refresh"></span> Loading' +
+    '</div>'
+  );
+
+  $.ajax({
+    url: apiSearchUrl( query, eiti.beckleyApiKey, {
+      size: size,
+      from: from
+    }),
+    cache: false,
+    dataType: 'json'
+  })
+    .done(function(json) {
+      $('.loading').remove();
+      $('.search-results-count').append( json.hits.total + ' search results');
+
+      if (json.hits.total == 0){
+        $('.search-no-results').show();
+      }
+      else{
+        $('.search-no-results').remove();
+      }
+
+      var hits = json.hits.hits.map(function(hit) {
+        var tags = (hit._source.tag || [])
+          .map(function(tag) {
+            return (
+              '<span class="search-result-list-tag">&nbsp;' +
+                '<a href="' + localSearchUrl(tag, {size: size, from: from}) + '" title="Search for ' + tag + '">' +
+                  tag +
+                '</a>&nbsp;' +
+              '</span>'
+            );
+          });
```

Comment:
When I work with arrays of things, I tend to prefer immutable maps over the arrays instead of string-building or array-building loops. Here we're actually preserving the map from tag to tagHTML on each item until the very end with the join( '' ).

in this case, we have a single constant binding between the source list and the output instead of a multi-step process of altering a given variable.